executor/insert_common.go

```diff
@@ -328,6 +341,19 @@ func (e *InsertValues) insertRowsFromSelect(ctx context.Context, exec func(rows
 	return nil
 }
 
+func (e *InsertValues) doBatchInsert() error {
+	sessVars := e.ctx.GetSessionVars()
+	e.ctx.StmtCommit()
```

Comment:
StmtCommit should be done only when the statement finish.
Batch insert during a statement is executing is not safe. @jackysp

Reply:
It must be done. Due to it is going to start a new transaction.

Reply:
ok, this function runs when `batchInsert` is set.